.travis.yml

```diff
@@ -34,8 +64,105 @@ install:
   # For now we disable it.
   - poetry config installer.parallel false
   - poetry install
+
+# Things to do before the script phase
 before_script:
-  - psql -U postgres -c 'create database nautobot;'
+  # If postgres: Create the database
+  - sh -c "if [ '$DB' = 'postgres' ]; then echo 'Creating PostgreSQL database...'; fi"
+  - sh -c "if [ '$DB' = 'postgres' ]; then psql -c 'CREATE DATABASE nautobot;' -U postgres; fi"
+  # If mysql: Upgrade/install MySQL 8.x, then create the database
+  - sh -c "if [ '$DB' = 'mysql' ]; then echo 'Upgrading MySQL to v8.x...'; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then wget https://repo.mysql.com//mysql-apt-config_0.8.17-1_all.deb; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo dpkg -i mysql-apt-config_0.8.17-1_all.deb; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo apt-get update -q; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo apt-get install -q -y --allow-unauthenticated -o Dpkg::Options::=--force-confnew mysql-server; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo systemctl restart mysql; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo mysql_upgrade; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then mysql --version; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then echo 'Creating MySQL database...'; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then mysql -e 'CREATE DATABASE IF NOT EXISTS nautobot;'; fi"
+  # Install the mysqlclient lib
+  - sh -c "if [ '$DB' = 'mysql' ]; then poetry install --extras mysql; fi"
+  # Install Docker Compose
   - pip install docker-compose
+
+# Script phase
 script:
-  - poetry run ./scripts/cibuild.sh
+  # Run unit tests
+  - invoke unittest --failfast --keepdb || travis_terminate 1
+  # Generate unit test coverage report
+  - invoke unittest-coverage || travis_terminate 1
+
+#
+## End "test" stage global config
+#
+
+# Job/stage matrix
+jobs:
+
+  # Terminate build matrix as soon as any job fails.
+  fast_finish: true
+
+  # Job definitions for custom stages
+  include:
+
+    - stage: lint
+
+      before_install:
+        - pip install poetry
+        # - pip install invoke requests black=="20.8b1" flake8=="3.9.2"  # Until there is a poetry install --dev-only install the linting tools here to speed up this step
+        - curl -Lo /home/travis/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.0.0/hadolint-Linux-x86_64
+        - chmod +x /home/travis/bin/hadolint
+
+      # Zero out the globals for this stage
+      # install: []  # nautobot needs to be installed for checking migrations
+      services: []
+      addons:
+        postgresql: ""
+      before_script:
+        - psql -U postgres -c 'CREATE DATABASE nautobot;'
```

Comment:
Yes. I'm not happy about that since without it linting can run in about 60s instead of 180s.

Reply:
A couple of ideas:
- is it any quicker if we use mysql instead of postgresql here?
- should we move the migrations check into the unittest section so that we don't need this here?